pkg/routeagent/controllers/route/route.go

```diff
@@ -66,6 +118,14 @@ func NewController(clusterID string, objectNamespace string, link *net.Interface
 		DeleteFunc: controller.handleRemovedEndpoint,
 	})
 
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: controller.enqueuePod,
+		UpdateFunc: func(old, new interface{}) {
+			controller.enqueuePod(new)
+		},
+		DeleteFunc: controller.handleRemovedPod,
```

Comment:
I know this pattern is used elsewhere in this code but it is not technically correct. If a delete occurred right after an add or update, the delete could be processed first. That's probably not likely to occur but we should be safe and ensure we process events in the order they are received.

Reply:
Actually looking at this further, we can't enqueue the _removes_ b/c we need access to the pod's state which we wouldn't be available in `processNextPod` since it only has the key.  Same with endpoints (sorry I missed that detail originally). Also it shouldn't matter if a delete is processed before an add b/c the `Get` in `processNextPod` would fail anyway.